utils/src/prompts.ts

```diff
@@ -0,0 +1,142 @@
+2. If a quote is for a non-native token, always use the getAllowance tool after getting a quote
+3. For allowance checks and approvals:
+   - ALWAYS use the exact amount from the quote's sellAmountCryptoPrecision field
+   - Convert the sellAmountCryptoPrecision to base units using toBaseUnit with the token's precision
+   - Compare the allowance with the exact base unit amount needed:
+     * If allowance >= required amount in base units, proceed with the swap
+     * If allowance < required amount in base units, request approval
+   - When requesting approval, use the exact base unit amount needed
+   - NEVER round or modify the amount for allowance/approval
+   - NEVER use a different amount than what's specified in the quote
+4. If an approval has been made, refetch a quote before continuing
+5. Before execution of a transaction, user always has to confirm they wish to proceed
+6. When sending transactions:
+   - ALWAYS use the complete transaction data from the quote response
+   - NEVER truncate or modify the transaction data
+   - Ensure the data field is a complete hex string
+   - Verify the data length is even (hex strings must have even length)`;
+
+// Formatting prompt
+const FORMATTING_PROMPT = `### Formatting Guidelines
+
+1. Make addresses clickable links with an emoji prefix
+2. Format responses in markdown, using backticks for code and addresses
+3. Use emojis appropriately to make the interaction more engaging
+4. Be concise but informative in responses`;
+
+// Math prompt
+const MATH_PROMPT = `### Math Instructions
+
+1. For unit conversions:
+   - Converting from base unit (wei) to human readable unit (eth): use fromBaseUnit
+   - Converting from human readable unit (eth) to base unit (wei): use toBaseUnit
+2. Always use the correct precision for each token:
+   - ETH/AVAX/BNB/xDAI: precision = 18
+   - USDC/USDT: precision = 6
+   - Other tokens: use precision from token search
+3. When handling amounts:
+   - ALWAYS maintain the exact precision throughout the entire process
+   - NEVER round or modify amounts when converting between units
+   - Use the exact amount from quotes for allowances and approvals
+   - Display amounts to users in human-readable format
+   - Use base units (with correct precision) for all transactions`;
+
+// Create message templates for each prompt
+const baseRoleTemplate = SystemMessagePromptTemplate.fromTemplate(BASE_ROLE);
+const balanceCheckTemplate =
+  SystemMessagePromptTemplate.fromTemplate(BALANCE_CHECK_PROMPT);
+const tokenSearchTemplate =
+  SystemMessagePromptTemplate.fromTemplate(TOKEN_SEARCH_PROMPT);
+const swapExecutionTemplate = SystemMessagePromptTemplate.fromTemplate(
+  SWAP_EXECUTION_PROMPT
+);
+const formattingTemplate =
+  SystemMessagePromptTemplate.fromTemplate(FORMATTING_PROMPT);
+const mathTemplate = SystemMessagePromptTemplate.fromTemplate(MATH_PROMPT);
+
+// Create a template that combines all prompts
+export const fullPromptTemplate = ChatPromptTemplate.fromMessages([
```

Comment:
As only `fullPromptTemplate` is consumed (and none of the sub-prompts), there should be no functional changes with this PR, yeah?